test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+void driver_test();
+void test_get_options();
+void reset_default_options(Options& o);
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+std::vector<std::pair<std::string, Options>> generate_cmdline_option_pairs();
+
+//unit test for the test driver
+void driver_test()
+{
+	test_get_options();
+}
+
+
+//unit test for get_options
+void test_get_options()
+{
+	//this is meant to replicate the main() parameter char* argv[]
+	//size 100 was chosen to ensure enough room for many options in a single command-line.
+	char* args[100];
+
+	const std::vector<std::pair<std::string, Options>> args_and_options = generate_cmdline_option_pairs();
+	std::size_t cmd_line_number{ 0 };
+	for (auto iter = args_and_options.begin(); iter != args_and_options.end(); ++iter)
+	{
+		std::vector<std::string> v = split(iter->first, ' ');
+		for (std::size_t i = 0; i < v.size(); ++i)
+			args[i] = v[i].data();
+
+		Options o = get_options(args, v.size());
+
+		compare(o, iter->second, "cmd_line_" + std::to_string(cmd_line_number));
+		++cmd_line_number;
+	}
+}
+
+
+//resets given Options& parameter data members to the default values
+void reset_default_options(Options& o)
+{
+	o.header = true;
+	o.summary = true;
+	o.header_text = "Running $suite";
+	o.prm = pass_report_mode::indicate;
+	o.fail_threshold = 0;
+	o.fom = file_open_mode::no_file;
+	o.output_filepath = "";
```

Comment:
can just call the default constructor? no need to assign a null value to it. L66 and L68